agent.go

```diff
@@ -1096,11 +1124,13 @@ func removeContainerCb(pod *pod, data []byte) error {
 		return fmt.Errorf("Container %s running, impossible to remove", payload.ID)
 	}
 
-	if err := pod.containers[payload.ID].removeContainer(payload.ID); err != nil {
+	if err := ctr.removeContainer(payload.ID); err != nil {
 		return err
 	}
 
+	pod.containersLock.Lock()
```

Comment:
This won't be enough here. Mutexes are a first step, but we also needs some synchronization waiting for all the started processes to end before we try to remove the container. Basically, this means we need to wait on a `sync` before to perform the container's removal here. Every time we create/start a new process, we should do something like `wg.Add(1)` and every time it ends, call into `wg.Done()`. This way we ensure we are not removing a container which is still used.